gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/JwtSecurity.java

```diff
@@ -299,20 +303,24 @@ public void onEvent(EurekaEvent event) {
                 }
 
                 events.add("Discovery Service Cache was updated.");
-                log.debug("Trying to reach the zOSMF.");
+                log.debug("Trying to reach the z/OSMF instance " + zosmfServiceId + ".");
                 if (providers.isZosmfAvailableAndOnline()) {
-                    events.add("zOSMF is avaiable and online.");
-                    log.debug("The zOSMF was reached ");
+                    events.add("z/OSMF instance " + zosmfServiceId + " is available and online.");
+                    log.debug("The z/OSMF instance {} was reached.", zosmfServiceId);
 
                     discoveryClient.unregisterEventListener(this); // only need to see zosmf up once to validate jwt secret
                     isZosmfReady = true;
 
                     try {
                         validateInitializationAgainstZosmf();
                     } catch (HttpsConfigError e) {
-                        apimlLog.log("org.zowe.apiml.gateway.jwtProducerConfigError", StringUtils.join(events, "\n"));
+                        synchronized (events) {
+                            apimlLog.log("org.zowe.apiml.gateway.jwtProducerConfigError", StringUtils.join(events, "\n"));
```

Comment:
to make it thread safe.

Reply:
Correct. the `events` Set is used in more than one thread. So it is just in case the other thread is still running when we iterate over the set to print out the content.